modules/publicize/class-async-publicize.php

```diff
@@ -0,0 +1,343 @@
+	 *
+	 * Used as a 'publicize_should_publicize_published_post' filter callback to
+	 * block publicize on posts that have been flagged with the
+	 * self::PUBLICIZE_BLOCK_META_KEY meta key, which is set
+	 * for posts that have been published via {@see 'flag_post_no_publicize()'}.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @see publish_wo_publicize()
+	 *
+	 * @param boolean $should_publicize True of post should be publicized.
+	 * @param  WP_Post $post Post instance in question.
+	 * @return boolean True if post should be published, false otherwise.
+	 */
+	public function can_publicize( $should_publicize, $post ) {
+		// Catch posts that are being published with asynchronous publicization and don't publicize them.
+		if ( metadata_exists( 'post', $post->ID, self::PUBLICIZE_BLOCK_META_KEY ) ) {
+			$meta_status = $this->check_async_meta_status( $post->ID );
+
+			if ( 'META_VALID' === $meta_status ) {
+				// Don't publicize since the flag has expired.
+				return false;
+			} else {
+				// Publicize post since the meta key has expired.
+				return true;
+			}
+		} else {
+			// Publicize post since it hasn't been flagged.
+			return true;
+		}
+	}
+
+	/**
+	 * Check if meta key flag for blocking publicize has expired.
+	 *
+	 * Implement the expiration of PUBLICIZE_BLOCK_META_KEY meta key.
+	 * Useful for handling the case where a post is flagged for no publicize
+	 * {@see flag_post_no_publicize()} but the publish operation does not
+	 * occur. Without a timeout, PUBLICIZE_BLOCK_META_KEY will be in effect
+	 * indefinitely, even if the post is later published from classic editor
+	 * or other non-Gutenberg source.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param int $post_id ID of post being queried.
+	 * @return string Returns 'META_VALID' if not expired, and 'META_EXPIRED' otherwise.
+	 */
+	public function check_async_meta_status( $post_id ) {
+		$time_flagged = get_post_meta( $post_id, self::PUBLICIZE_BLOCK_META_KEY, true );
+
+		if ( is_numeric( $time_flagged ) && ( $time_flagged < time() + self::META_KEY_TIMEOUT_SECONDS ) ) {
+			return 'META_VALID';
+		}
+
+		return 'META_EXPIRED';
+	}
+
+	/**
+	 * REST api callback for publishing a post without publicizing it.
+	 *
+	 * Function exposed as endpoint 'publicize/posts/<post_id>/flag-no-publicize'.
+	 * Marks a post (using post meta key) so that it is NOT later publicized.
+	 * This should be called if the caller is about to publish a post but does not
+	 * want it automatically publicized.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @see __construct()
+	 *
+	 * @param WP_REST_Request $request Request instance from REST call.
+	 * @return string|null Result body to return to REST caller
+	 */
+	public function flag_post_no_publicize( $request ) {
+		$post_id = $request['post_id'];
+
+		if ( ! current_user_can( 'publish_post', $post_id ) ) {
+			return 'Current user cannot publish this post';
+		}
+
+		// If post does not exist.
+		if ( get_post_status( $post_id ) === false ) {
```

Comment:
You'll want to use Yoda conditons, as per the WP coding standards:
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions